libraries/ESP8266WiFi/src/CertStoreSPIFFSBearSSL.cpp

```diff
@@ -123,35 +125,42 @@ const br_x509_trust_anchor *CertStoreSPIFFSBearSSL::findHashedTA(void *hashed_dn
       }
       if (d.read(der, der_len) != der_len) {
         d.close();
+        free(der);
         return nullptr;
       }
       d.close();
 
-      br_x509_decoder_context dc;
-      br_x509_decoder_init(&dc, dn_append_null, nullptr, nullptr, nullptr);
-      br_x509_decoder_push(&dc, der, der_len);
+      br_x509_decoder_context *dc = new br_x509_decoder_context;
+      br_x509_decoder_init(dc, dn_append_null, nullptr, nullptr, nullptr);
+      br_x509_decoder_push(dc, der, der_len);
       free(der);
-      br_x509_pkey *pk = br_x509_decoder_get_pkey(&dc);
+      br_x509_pkey *pk = br_x509_decoder_get_pkey(dc);
       if (!pk) {
+        delete dc;
         return nullptr;
       }
 
       br_x509_trust_anchor *ta = (br_x509_trust_anchor*)malloc(sizeof(br_x509_trust_anchor));
       if (!ta) {
+        delete dc;
         return nullptr;
       }
       memset(ta, 0, sizeof(*ta));
       ta->dn.data = (uint8_t*)malloc(sizeof(ci.sha256));
       if (!ta->dn.data) {
         free(ta);
+        delete dc;
         return nullptr;
       }
       memcpy(ta->dn.data, ci.sha256, sizeof(ci.sha256));
       ta->dn.len = sizeof(ci.sha256);
       ta->flags = 0;
-      if (br_x509_decoder_isCA(&dc)) {
+      if (br_x509_decoder_isCA(dc)) {
         ta->flags |= BR_X509_TA_CA;
       }
+
+      delete dc; // Done with x509 decoder
```

Comment:
At this point `pk` still holds reference to structure inside `dc`, so it cannot be freed here.
Issure became visible when add some debug prints after this line corrupted the private key.